File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/Flag.java
package com.nopalsoft.ponyrace.game_objects;

import com.nopalsoft.ponyrace.game.TileMapHandler;

import java.util.Random;

public class Flag {
    public State state;
    public ActionType actionType;
    Random random;

    public Flag(TileMapHandler world, ActionType actionType) {
        random = world.random;
        state = State.NORMAL;
        this.actionType = actionType;
    }

    public boolean permitirSalto() {
        if (state == State.NORMAL && random.nextBoolean()) {
            state = State.ACTIVE;
            return true;
        }
        return false;
    }

    public enum State {
        NORMAL,
        ACTIVE
    }

    public enum ActionType {
        JUMP_LEFT,
        JUMP_RIGHT,
        JUMP
    }
}

File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/Flag.kt
package com.nopalsoft.ponyrace.game_objects

import com.nopalsoft.ponyrace.game.TileMapHandler
import java.util.Random

class Flag(world: TileMapHandler, @JvmField var actionType: ActionType?) {

    @JvmField
    var state: State? = State.NORMAL
    var random: Random = world.random

    fun permitirSalto(): Boolean {
        if (state == State.NORMAL && random.nextBoolean()) {
            state = State.ACTIVE
            return true
        }
        return false
    }

    enum class State {
        NORMAL,
        ACTIVE
    }

    enum class ActionType {
        JUMP_LEFT,
        JUMP_RIGHT,
        JUMP
    }
}
